sdk/cognitiveservices/Knowledge.QnAMaker/src/Customizations/EndpointKeyServiceClientCredentials.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Microsoft.Azure.CognitiveServices.Knowledge.QnAMaker
+{
+    using System;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Rest;
+
+    /// <summary>
+    /// Allows authentication to the API using a basic apiKey mechanism
+    /// </summary>
+    public class EndpointKeyServiceClientCredentials : ServiceClientCredentials
+    {
+        private readonly string endpointKey;
+
+        /// <summary>
+        /// Creates a new instance of the EndpointKeyServiceClientCredentials class
+        /// </summary>
+        /// <param name="endpointKey">The Endpoint key to authenticate and authorize as</param>
+        public EndpointKeyServiceClientCredentials(string endpointKey)
+        {
+            this.endpointKey = endpointKey;
+        }
+
+        /// <summary>
+        /// Add the Basic Authentication Header to each outgoing request
+        /// </summary>
+        /// <param name="request">The outgoing request</param>
+        /// <param name="cancellationToken">A token to cancel the operation</param>
+        public override Task ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+                throw new ArgumentNullException("request");
+
+            request.Headers.Add("Authorization", $"EndpointKey {this.endpointKey}");
+
+            return Task.FromResult<object>(null);
```

Comment:
Why do this?

Reply:
QnAMaker runtime needs Authorization header to have scheme as EndpointKey

Reply:
I was wondering about the `return Task.FromResult<object>(null);` part

Reply:
I Tried Task.CompletedTask but that doesn't work for net452

Reply:
What are you trying to accomplish with the return?

Reply:
This is needed since that is the signature of the method we are overriding in the base class.

Reply:
Will the return type of this method be consumed anywhere else? If it does, returning null will end up with a nullref exception at a later, random time.

That is why I was wondering why we return `null` here (as well as why the wonky Task creation 😄 - but that wonkiness makes sense now ).

Reply:
The return type is just Task (which is not null). The reason this works is because Task<T> derives from Task.